src/topology/sheaves/skyscraper.lean

```diff
@@ -0,0 +1,169 @@
+-/
+
+noncomputable theory
+
+open topological_space
+open category_theory category_theory.limits
+open Top
+open opposite
+
+universes u v w
+
+variables {X : Top.{u}} (p₀ : X) [Π (U : opens X), decidable (p₀ ∈ U)]
+
+section
+
+variables {C : Type v} [category.{w} C] (A : C) [has_terminal C]
+
+/--
+A skyscraper presheaf is a presheaf supported at a single point: if `p₀ ∈ X` is a specified
+point, then the skyscraper presheaf `𝓕` with value `A` is defined by `U ↦ A` if `p₀ ∈ U` and
+`U ↦ *` if `p₀ ∉ A` where `*` is some terminal object.
+-/
+@[simps] def skyscraper_presheaf : presheaf C X :=
+{ obj := λ U, if p₀ ∈ unop U then A else terminal C,
+  map := λ U V i, if h : p₀ ∈ unop V
+    then eq_to_hom $ by erw [if_pos h, if_pos (le_of_hom i.unop h)]
+    else ((if_neg h).symm.rec terminal_is_terminal).from _,
+  map_id' := λ U,
+  begin
+    split_ifs,
+    { apply eq_to_hom_refl },
+    { exact ((if_neg h).symm.rec terminal_is_terminal).hom_ext _ _ },
+  end,
+  map_comp' := λ U V W iVU iWV,
+  begin
+    by_cases hW : p₀ ∈ unop W,
+    { have hV : p₀ ∈ unop V := le_of_hom iWV.unop hW,
+      simp only [dif_pos hW, dif_pos hV, eq_to_hom_trans] },
+    { rw [dif_neg hW], apply ((if_neg hW).symm.rec terminal_is_terminal).hom_ext }
+  end }
+
+end
+
+section
+
+-- In this section, we calculate the stalks for skyscraper presheaves.
+-- We need to restrict universe level.
+
+variables {C : Type v} [category.{u} C] (A : C) [has_terminal C]
+
+/--
+The cocone at `A` for the stalk functor of `skyscraper_presheaf p₀ A` when `y ∈ closure {p₀}`
+-/
+@[simps] def skyscraper_presheaf_cocone_of_specializes {y : X} (h : p₀ ⤳ y) :
+  cocone ((open_nhds.inclusion y).op ⋙ skyscraper_presheaf p₀ A) :=
+{ X := A,
+  ι := { app := λ U, eq_to_hom $ if_pos $ h.mem_open U.unop.1.2 U.unop.2,
+    naturality' := λ U V inc, begin
+      change dite _ _ _ ≫ _ = _, rw dif_pos,
+      { erw [category.comp_id, eq_to_hom_trans], refl },
+      { exact h.mem_open V.unop.1.2 V.unop.2 },
+    end } }
+
+/--
+The cocone at `A` for the stalk functor of `skyscraper_presheaf p₀ A` when `y ∈ closure {p₀}` is a
+colimit
+-/
+noncomputable def skyscraper_presheaf_cocone_is_colimit_of_specializes
+  {y : X} (h : p₀ ⤳ y) : is_colimit (skyscraper_presheaf_cocone_of_specializes p₀ A h) :=
+{ desc := λ c, eq_to_hom (if_pos trivial).symm ≫ c.ι.app (op ⊤),
+  fac' := λ c U, begin
+    rw ← c.w (hom_of_le $ (le_top : unop U ≤ _)).op,
+    change _ ≫ _ ≫ dite _ _ _ ≫ _ = _,
+    rw dif_pos,
+    { simpa only [skyscraper_presheaf_cocone_of_specializes_ι_app,
+        eq_to_hom_trans_assoc, eq_to_hom_refl, category.id_comp] },
+    { exact h.mem_open U.unop.1.2 U.unop.2 },
+  end,
+  uniq' := λ c f h, by rw [← h, skyscraper_presheaf_cocone_of_specializes_ι_app,
+    eq_to_hom_trans_assoc, eq_to_hom_refl, category.id_comp] }
+
+/--
+If `y ∈ closure {p₀}`, then the stalk of `skyscraper_presheaf p₀ A` at `y` is `A`.
+-/
+@[reducible]
```

Comment:
Are you sure about the `@[reducible]` tag? I am not against it, just wondering why it is there.

Reply:
I have removed `@[reducible]`, I have forgot why I put it there and removing it has no side effects.